Tests/Command/ConsumeCommandTest.php

```diff
@@ -16,28 +20,93 @@ class ConsumeCommandTest extends KernelTestCase
 
     protected $mockConsumer;
 
-    public function setMockConsumer($expirationCount)
+    /**
+     * @var MockObject
+     */
+    private $mockQueueDriver;
+
+    public function setUp()
     {
         self::bootKernel();
+    }
+
+    public function setMockQueueDriver()
+    {
+        $msg = new QueueMessage(new Message());
+
+        $this->mockQueueDriver = $this
+            ->getMockBuilder(QueueDriverInterface::class)
+            ->setMethods([
+                'configure',
+                'connect',
+                'disconnect',
+                'isConnected',
+                'send',
+                'ack',
+                'nack',
+                'createQueueMessage',
+                'setFormat',
+                'getFormat',
+                'getInternalType'
+            ])
+            ->getMock();
+        $this->mockQueueDriver
+            ->method('configure');
+        $this->mockQueueDriver
+            ->method('connect');
+        $this->mockQueueDriver
+            ->method('disconnect');
+        $this->mockQueueDriver
+            ->method('isConnected');
+        $this->mockQueueDriver
+            ->method('send')
+            ->with($msg);
+        $this->mockQueueDriver
+            ->method('ack');
+        $this->mockQueueDriver
+            ->method('nack');
+        $this->mockQueueDriver
+            ->method('createQueueMessage');
+        $this->mockQueueDriver
+            ->method('setFormat');
+        $this->mockQueueDriver
+            ->method('getFormat');
+        $this->mockQueueDriver
+            ->method('getInternalType');
+
+        self::$kernel->getContainer()->set(QueueDriverInterface::class, $this->mockQueueDriver);
+        self::$kernel->getContainer()->set('doctrine', $this->createMock(RegistryInterface::class));
```

Comment:
If the method is called `setMockQueueDriver` why do we need to mock 'doctrine'?

Reply:
To keep compatibility with the symfony 2.8 that seems to call doctrine under the woods.

Reply:
Because of this config: https://github.com/smartboxgroup/integration-framework-bundle/blob/master/Tests/App/config/services.yml#L4